scripts/generateAssets.ts

```diff
@@ -0,0 +1,114 @@
+import path from "path";
+import semver from "semver";
+
+import { promises as fs } from "fs";
+
+/** Representation of History File JSON */
+type HistoryFile = {
+  [packageName: string]: Array<{
+    date: string;
+    versions: Record<string, number>;
+  }>;
+};
+
+/** Representation of packed asset file */
+type AssetHistoryFile = { points: AssetHistoryPoint[] };
+type AssetHistoryPoint = { date: number; version: string; count: number };
+
+/** Minimum number of downloads for a version to be recored */
+
+/** Maximum datepoints per package to keep in webpage assets */
+const maxAssetHistoryEntries = 60;
+
+(async () => {
+  try {
+    await generateWebpageAssets();
+  } catch (e) {
+    console.error(e);
+    process.exit(1);
+  }
+})();
+
+/**
+ * Generate reduced JSON to power the webpage graph
+ */
+async function generateWebpageAssets() {
+  const fullHistory: HistoryFile = require(fullHistoryPath());
+
+  for (const [packageName, counts] of Object.entries(fullHistory)) {
+    const trimmedHistory = counts.slice(0, maxAssetHistoryEntries);
+    const datePoints: AssetHistoryPoint[] = [];
+
+    for (const fileDatePoint of trimmedHistory) {
+      for (const [version, count] of Object.entries(fileDatePoint.versions)) {
+        datePoints.push({
+          date: new Date(fileDatePoint.date).getTime(),
+          version,
+          count,
+        });
+      }
+    }
+
+    const sortedPoints: AssetHistoryPoint[] = datePoints.sort(
+      compareAssetHistoryPoint
+    );
+
+    const historyAssetPath = path.join(
+      __dirname,
+      "..",
+      "src/generated_assets",
+      `${packageName.replace(/\//g, "_")}.json`
+    );
+
+    const historyFile: AssetHistoryFile = { points: sortedPoints };
+    await fs.writeFile(historyAssetPath, JSON.stringify(historyFile));
```

Comment:
Do we really need to await here?

Reply:
We import the promise based version of `fs` above:

```ts
import { promises as fs } from "fs";
```

We could write concurrently to give up sequentiality, but I'd kind of rather this be sequential, to be easier to reason about.

Reply:
Sorry, what I meant is that I think you can fire-and-forget in this case. It doesn't look like the rest of the logic needs to wait till the write is done.